Code/ForceField/Wrap/ForceField.cpp

```diff
@@ -130,6 +131,64 @@ PyObject *ForceFieldGetExtraPointLoc(PyForceField *self, unsigned int idx) {
   return res;
 }
 
+double PyForceField::calcEnergyWithPos(const python::object &pos) {
+  PRECONDITION(this->field, "no force field");
+  if (pos != python::object()) {
+    const python::list &posList = reinterpret_cast<const python::list &>(pos);
+    long int ll = python::len(posList);
+    PRECONDITION(ll == 3 * this->field->numPoints(),
+      "The list must have length equal to 3 * NumPoints()");
+    boost::shared_array<double> c(new double[ll]);
+    for (long int i = 0; i < ll; ++i)
+      c[i] = boost::python::extract<double>(posList[i]);
+    return this->field->calcEnergy(c.get());
+  }
+  else
+    return this->field->calcEnergy();
+}
+
+PyObject *PyForceField::positions() {
+  PRECONDITION(this->field, "no force field");
+  long int s = 3 * this->field->numPoints();
+  PyObject *coordTuple = PyTuple_New(s);
+  const RDGeom::PointPtrVect &p = this->field->positions();
+  int i = 0;
+  PyObject *coordItem;
+  for (RDGeom::PointPtrVect::const_iterator it = p.begin(); it != p.end(); ++it) {
+    coordItem = PyFloat_FromDouble(static_cast<RDGeom::Point3D *>(*it)->x);
+    PyTuple_SetItem(coordTuple, i++, coordItem);
+    coordItem = PyFloat_FromDouble(static_cast<RDGeom::Point3D *>(*it)->y);
+    PyTuple_SetItem(coordTuple, i++, coordItem);
+    coordItem = PyFloat_FromDouble(static_cast<RDGeom::Point3D *>(*it)->z);
+    PyTuple_SetItem(coordTuple, i++, coordItem);
+  }
+  return coordTuple;
+}
+
+PyObject *PyForceField::calcGradWithPos(const python::object &pos) {
+  PRECONDITION(this->field, "no force field");
+  long int s = 3 * this->field->numPoints();
+  boost::shared_array<double> g(new double[s]);
```

Comment:
Why use a shared_array here, why not a std::vector<double>?

Reply:
I guess because I haven't yet metabolised that the RDKit has switched to C++11 :-/